classes/Project/Project.mjs

```diff
@@ -45,7 +50,39 @@ export default class Project {
     return database.remove(projectId)
   }
 
-  checkUserAccess(userAgent) {
+  async addMember(email, rolesString) {
+    try {
+      let user = await User.getByEmail(email)
+      const roles = this.parseRoles(rolesString)
+      let updatedProject
+      let message = `You have been invited to the TPEN project ${
+        this.projectData?.name
+      } with the following role(s): ${roles.join(", ")}.`
+
+      if (user) {
+        updatedProject = await this.inviteExistingTPENUser(user, roles)
+      } else {
+        let {newUser, projectData} = await this.inviteNewTPENUser(email, roles)
+        // We will replace this URL with the correct url
+        const url = `https://cubap.auth0.com/u/signup?invite-code=${newUser.inviteCode}`
+        updatedProject = projectData
+        user = newUser
+        message += `<p>Click the button below to get started with your project</p> 
+        <button class = "buttonStyle" ><a href=${url} >Get Started</a> </button>
+        or copy the following link into your web browser <a href=${url}>${url}</a> </p>`
+      }
+
+      await sendMail(user, `Invitation to ${this.projectData?.name}`, message)
```

Comment:
We probably don't need to wait for this.

Reply:
even though `sendMail()` is labelled `async` the Transporter.send method isn't, so I don't think there is actually a Promise to generate here.

Reply:
The `err(err)` is a callback as I read it, in nodemailer, so that's where we could be rejecting or throwing to reach the `catch` block below.